cmds/core/io/smn.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 the u-root Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build amd64
+
+package main
+
+import (
+	"github.com/u-root/u-root/pkg/memio"
+)
+
+type op func() error
+
+// This is a const at present but there are no guarantees.
+const pcibase = 0xe0000000
+
+func init() {
+	usageMsg += `io rs index # read from system management network
+io ws index value # write value to system management network
+`
+	addCmd(readCmds, "rs", &cmd{smnRead, 32, 32})
+	addCmd(writeCmds, "ws", &cmd{smnWrite, 32, 32})
+}
+
+func do(addr int64, data memio.UintN, op func(int64, memio.UintN) error) error {
+	var a = newInt(uint64(addr), 32)
+	if err := memio.Write(pcibase+0xb8, a); err != nil {
```

Comment:
The build fails because plan9 doesn't have this function memio.Write.

```
2020/12/23 21:15:27 Build environment: GOARCH=amd64 GOOS=plan9 GOROOT=/usr/local/go GOPATH=/go CGO_ENABLED=0 GO111MODULE=off
2020/12/23 21:15:27 GOOS is not linux. Did you mean to set GOOS=linux?
2020/12/23 21:15:31 error building: error building go package in "/go/src/github.com/u-root/u-root/bb": # github.com/u-root/u-root/cmds/core/io/.bb
../cmds/core/io/.bb/smn.go:28:12: undefined: memio.Write
../cmds/core/io/.bb/smn.go:35:24: undefined: memio.Write
../cmds/core/io/.bb/smn.go:39:24: undefined: memio.Read
```